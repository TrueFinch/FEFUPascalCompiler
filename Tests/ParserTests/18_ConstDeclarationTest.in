program a;
Const hello = 'Hello, world!';
      n = 5;
var
    i: integer = 0;
begin
    for i := 0 to n - 1 do
        writeln(hello);
end.