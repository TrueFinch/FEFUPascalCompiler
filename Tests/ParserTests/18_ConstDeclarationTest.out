                                    a                                      
                                    ┼                                      
                                MainBlock                                  
             ┌──────────────────────┴──────────────────────┐               
           const                   var             CompoundStatement       
          ┌──┴──────────┐           ┼                      ┼               
          =             =      InitVarDecl            ForStatement         
   ┌──────┴───┐        ┌┴─┐  ┌──────┴───────┐  ┌─────┬─────┴─────┐         
 hello  Hello, world!  n  5  i  SimpleType  0  i    to      FunctionCall   
                                   ┌┴             ┌──┴┐       ┌──┴────┐    
                                integer           0   -    writeln  hello  
                                                     ┌┴─┐                  
                                                     n  1                  